default/python/AI/ProductionAI.py

```diff
@@ -1948,3 +1945,69 @@ def _build_translator():
             break
     return turn_cost
     # may_enqueue_for_stability? Building is rather expensive...
+
+
+def _build_regional_administration() -> float:
+    """Consider building Imperial Regional Administrations, returns added turn costs"""
+    building_type = BuildingType.REGIONAL_ADMIN
+    current_admin_sys = building_type.built_at_sys(True) | BuildingType.PALACE.built_at_sys(True)
+    # No admins at all means no palace. If we cannot even find a place for the palace,
+    # there is no point in building regional admins
+    if not building_type.available() or not current_admin_sys:
+        return 0.0
+    universe = fo.getUniverse()
+    jumps_to_admin = [
+        (min(universe.jumpDistance(sys_id, admin) for admin in current_admin_sys), sys_id)
+        for sys_id in get_owned_planets()
+    ]
+    jumps_to_admin.sort(reverse=True)
+    # Currently 6 is required, this is supposed to change. Note that the game allows to enqueue several ones
+    # close to each other, but only one will get finished.
+    if jumps_to_admin[0][0] < 6:
+        return 0.0
+    debug(f"current_admin_sys: {current_admin_sys}, jumps_to_admin = {jumps_to_admin}")
```

Comment:
admin and sys area bit confusing, let's use full words.  

As I understand this is a candidate system for the administration that is located {jumps} from another administration.

Reply:
I improved the variable names.
I'll stick to admin because it is used a lot all over the function and it should be rather obvious what it stands for.